chain-db-watcher/src/graphql_helpers.ts

```diff
@@ -1,37 +1,140 @@
+import chalk from 'chalk';
 import dotenv from 'dotenv';
 import { GraphQLClient } from 'graphql-request';
 
-import { addPostAndProposalMutation, getProposalQuery, loginMutation } from './queries';
+import {
+	addPostAndProposalMutation,
+	addReferendumIdMutation,
+	getProposalQuery,
+	getProposalWithNullReferendumQuery,
+	getTabledProposalsAtBlockQuery,
+	loginMutation
+} from './queries';
 
 dotenv.config();
 
-const graphqlServerUrl = process.env.REACT_APP_HASURA_GRAPHQL_URL;
+const discussionGraphqlUrl = process.env.REACT_APP_HASURA_GRAPHQL_URL;
+const democracyTopicId = process.env.DEMOCRACY_TOPIC_ID;
+const proposalPostTypeId = process.env.HASURA_PROPOSAL_POST_TYPE_ID;
+const botProposalUserId = process.env.BOT_PROPOSAL_USER_ID;
+const authServerUrl = process.env.AUTH_SERVER_URL;
+const chainGraphqlServerUrl = process.env.CHAIN_DB_GRAPHQL_URL;
+
+export const getToken = async (): Promise<string> => {
+	const credentials = {
+		username: process.env.PROPOSAL_BOT_USERNAME,
+		password: process.env.PROPOSAL_BOT_PASSWORD
+	};
+
+	if (!authServerUrl) {
+		throw new Error('Auth server url not set in .env file.');
+	}
+
+	if (!credentials.username || !credentials.password) {
+		throw new Error(
+			"PROPOSAL_BOT_USERNAME or PROPOSAL_BOT_PASSWORD environment variables haven't been set for the proposal bot to login."
+		);
+	}
+
+	try {
+		const client = new GraphQLClient(authServerUrl, { headers: {} });
+
+		return client
```

Comment:
don't mix async await and then catch

you can do 

const data = await client.request()

try catch with catch the error

Reply:
done